### ISSUE TITLE
Reproduce NCC loss

### ISSUE THREAD

**Issue by John1231983** [voxelmorph/voxelmorph#92]
Hello, this is not an issue. I just want to ask you about the NCC loss in CVPR 2018, that I want to reproduce 

1. For two following lines, if we use in 3D image, then win=[9,9,9], strides =[1,1,1] and  sum_filt = ones([9,9,9,1,1]). So, if my input has 2 channels, then the sum_fit must be size of `9x9x9x2x1`, is it right?

https://github.com/voxelmorph/voxelmorph/blob/c975b94d0ef6d235afe39a1b49f28e5f10861631/src/losses.py#L43
https://github.com/voxelmorph/voxelmorph/blob/c975b94d0ef6d235afe39a1b49f28e5f10861631/src/losses.py#L54
https://github.com/voxelmorph/voxelmorph/blob/c975b94d0ef6d235afe39a1b49f28e5f10861631/src/losses.py#L57

This is my reproduce code
```
    def ncc_loss(self, I, J):        
        eps = 1e-5
        n = 9
        win = [n] *3 # [9,9,9]
        # compute CC squares
        I2 = torch.mul(I,I)
        J2 = torch.mul(J,J)
        IJ = torch.mul(I,J)
        # compute filters
        batch_size, channels, _, _, _ = I.shape
        sum_filt = torch.ones((1, channels, n, n, n)).float().to('cuda')
        strides = [1]*3 # [1,1,1]
        # compute local sums via convolution
        I_sum = F.conv3d(I, sum_filt, stride= strides, padding=1)
        J_sum = F.conv3d(J, sum_filt, stride=strides, padding=1)
        I2_sum = F.conv3d(I2, sum_filt, stride=strides, padding=1)
        J2_sum = F.conv3d(J2, sum_filt, stride=strides, padding=1)
        IJ_sum = F.conv3d(IJ, sum_filt, stride=strides, padding=1)
        # compute cross correlation
        win_size = np.prod(win)
        u_I = I_sum / win_size
        u_J = J_sum / win_size
        cross = IJ_sum - u_J*I_sum - u_I*J_sum + u_I*u_J*win_size
        I_var = I2_sum - 2 * u_I * I_sum + u_I*u_I*win_size
        J_var = J2_sum - 2 * u_J * J_sum + u_J*u_J*win_size
        cc = cross*cross / (I_var*J_var + eps)        
        return 1 - torch.mean(cc)

```

However, I got some big value in the loss during training (i.e. ncc loss: -162.68817  in epoch 3). Could you guess any issue in the code?
```
Epoch: (  1)  | Total loss: 0.83465 |ncc loss: 0.59442 |  Smooth loss 0.24022
Epoch: (  1)  | Total loss: 0.81657 |ncc loss: 0.57388 |  Smooth loss 0.24269
Epoch: (  1) | Total loss: 0.87251 |ncc loss: 0.62968 |  Smooth loss 0.24282
Epoch: (  1) | Total loss: 0.89325 |ncc loss: 0.65125 |  Smooth loss 0.24199
...
Epoch: (  3)  | Total loss: -0.01893 |ncc loss: -0.25998 |  Smooth loss 0.24104
Epoch: (  3)  | Total loss: 0.73279 |ncc loss: 0.49024 |  Smooth loss 0.24255
Epoch: (  3)  | Total loss: -162.44589 |ncc loss: -162.68817 |  Smooth loss 0.24228
Epoch: (  3)  | Total loss: 1.79963 |ncc loss: 1.55769 |  Smooth loss 0.24194
Epoch: (  3) | Total loss: 0.51922 |ncc loss: 0.27774 |  Smooth loss 0.24147
Epoch: (  3) | Total loss: 0.00790 |ncc loss: -0.23127 |  Smooth loss 0.23916
```
